packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -420,11 +438,23 @@ export class TreeCheckout implements ITreeCheckoutFork {
 				this.events.emit("afterBatch");
 			}
 			if (event.type === "replace" && getChangeReplaceType(event) === "transactionCommit") {
+				// commit the in progress removed roots if there are no more transactions
+				if (!this.transaction.inProgress()) {
+					assert(
+						this.removedRootsInTransaction !== undefined,
+						"a transaction should operate on a copy of the removed roots index",
+					);
+					this.removedRoots.loadIndex(this.removedRootsInTransaction);
```

Comment:
Transactions being aborted is fairly rare, so I think we should optimize for the case where the transaction is committed, which would mean avoiding a deep copy in the commit case. Is there something that motivated you to adopt this approach rather than the one I had recommended in our Teams conversation?
(Here's the relevant excerpt: "We could imagine an approach where we only keep the latest but take a snapshot of our data at the start of a transaction. If the transaction is aborted then we revert to that snapshot, and if the transaction is completed then we discard the snapshot.")

Reply:
I think I just misunderstood you. How can I take a snapshot?

Reply:
`const preTransactionSnapshot = this.removedRoots.clone();`

Reply:
Oh I see, I just mixed up what's being kept when.

Reply:
You don't have to use the term "snapshot" for this though, since that would be a bit of an overload.

Reply:
Oh, I ended up using snapshot. Lmk if you think of a better term.